integration/backward_compatibility_test.go

```diff
@@ -120,3 +128,85 @@ func runBackwardCompatibilityTestWithChunksStorage(t *testing.T, previousImage s
 		require.NoError(t, s.Stop(querier))
 	}
 }
+
+// Check for issues like https://github.com/cortexproject/cortex/issues/2356
+func runNewDistributorsCanPushToOldIngesters(t *testing.T, previousImage string) {
```

Comment:
Note to myself: the case "distributors can push to old ingesters" is actually covered by `runBackwardCompatibilityTestWithChunksStorage()`. The real difference in the setup is the replication factor.